t/unit/utils/test_collections.py

```diff
@@ -293,7 +293,7 @@ def test_update(self):
 
     def test_iterable_and_ordering(self):
         s = LimitedSet(maxlen=35, expires=None)
-        # we use a custom clock here, as time.time() does not have enough
+        # we use a custom clock here, as monotonic() does not have enough
```

Comment:
Almost seems like this test isn't necessary at all any more at all.  You can't get any closer to the real time other than monotonic().  I could sub in the **s.add(i, now=next(clock))** to just **s.add(i, now=monotonic())** ([reference](https://github.com/celery/celery/blob/2394e738a7d0c0c736f5d689de4d32325ba54f48/t/unit/utils/test_collections.py#L298-L300)), but that would just be re-iterating what is under the hood anyway.

Or are you suggesting that if we detect two (specified) times that are exactly the same, we add **1e9** to it (instead of **1e6**)? It would be a bit like merging the original pull request with this one.  This test function you're referencing here could then pass in **time()**? Might be overkill at this point though; thoughts?